lib/fluent/plugin/kubernetes_metadata_watch_namespaces.rb

```diff
@@ -47,29 +47,37 @@ def set_up_namespace_thread
         log.info('410 Gone encountered. Restarting namespace watch to reset resource versions.', e)
         namespace_watcher = nil
       rescue StandardError => e
-        @stats.bump(:namespace_watch_failures)
-        if Thread.current[:namespace_watch_retry_count] < @watch_retry_max_times
-          # Instead of raising exceptions and crashing Fluentd, swallow
-          # the exception and reset the watcher.
-          log.info(
-            'Exception encountered parsing namespace watch event. ' \
-            'The connection might have been closed. Sleeping for ' \
-            "#{Thread.current[:namespace_watch_retry_backoff_interval]} " \
-            'seconds and resetting the namespace watcher.', e
-          )
-          sleep(Thread.current[:namespace_watch_retry_backoff_interval])
-          Thread.current[:namespace_watch_retry_count] += 1
-          Thread.current[:namespace_watch_retry_backoff_interval] *= @watch_retry_exponential_backoff_base
+        if e.message == "Unauthorized"
```

Comment:
same question here regarding evaluating against a known code instead of a string